EIPS/eip-6806.md

```diff
@@ -0,0 +1,98 @@
+  function setWhitelistedContract(address whitelistContract, bool ignoreReset) external;
+}
+```
+
+**Functions**
+
+### getHoldingInfo
+
+```
+function getHoldingInfo(uint256 tokenId) external view returns (address holder, uint256 holdingTime);
+```
+
+This function returns the current holder of the specified NFT and the length of time (in seconds) the NFT has been held by the current account.
+
+* `tokenId`: The unique identifier of the NFT.
+* Returns: A tuple containing the current holder's address and the holding time (in seconds).
+
+### setWhitelistedContract
+
+```
+function setWhitelistedContract(address whitelistContract, bool ignoreReset) external;
+```
+
+This function allows the contract owner or an authorized account to specify whether a specific transfer should be ignored for the purposes of tracking holding time.
+
+* `whitelistContract`: The smart contract address that should be whitelisted for holding time reset exceptions. 
+* `ignoreReset`: A boolean value indicating whether the holding time reset should be ignored (`true`) or not (`false`) when transferring to or from the specified whitelistContract. 
+
+## Rationale
+
+The addition of the `getHoldingInfo` and `setWhitelistedContract` functions to an extension of the ERC-721 standard enables developers to implement NFT-based applications that require holding time information and allow for selective transfer exceptions. This extension maintains compatibility with existing ERC-721 implementations while offering additional functionality for new use cases.
+
+The `getHoldingInfo` function provides a straightforward method for retrieving the holding time and holder address of an NFT. By using seconds as the unit of time for holding duration, it ensures precision and compatibility with other time-based functions in smart contracts.
+
+The `setWhitelistedContract` function introduces flexibility in determining which transfers should affect holding time calculations. By allowing the contract owner or authorized accounts to whitelist specific addresses, transfers involving these addresses will not reset the holding time. This is particularly important for NFT-based financial transactions or special cases where holding time should not be affected by transfers.
+
+Together, these functions enhance the utility of the ERC-721 standard, enabling developers to create more sophisticated applications and experiences based on NFT holding time and offering the ability to selectively ignore holding time resets during transfers to or from whitelisted addresses. 
+
+## Backwards Compatibility
+
+This proposal is fully backwards compatible with the existing ERC-721 standard, as it extends the standard with new functions that do not affect the core functionality.
+
+## Reference Implementation 
+
+An implementation of this EIP will be provided upon acceptance of the proposal.
+
+## Security Considerations
```

Comment:
The content of this section seems boilerplate generated by ChatGPT since it appear to be unrelated to the actual technicality of providing holding time